src/compiler/marks.ts

```diff
@@ -357,8 +357,53 @@ export namespace bar {
   }
 
   export function labels(model: Model) {
-    // TODO(#64): fill this method
-    return undefined;
+    // TODO(#64):
+    // √ horizontal support
+    // - vertical support
+    // - color support
+    // - container height/stretching consideration
+    // - consider defaults for offset
+
+    var l: any = {};
+    const orient = model.marksConfig('orient');
+
+    l.align = { value: (orient !== 'horizontal') ? "center" : "left" };
+
+    l.text = { field: model.has(LABEL) ? model.field(LABEL) : ((orient !== 'horizontal') ? model.field(Y) : model.field(X)) };
```

Comment:
if there is not `label` channel, this function would not be evaluated at all.   

See [this line](https://github.com/vega/vega-lite/blob/master/src/compiler/marks.ts#L104)